Training/Training/Training.c

```diff
@@ -1,14 +1,194 @@
+/// </summary>
+int NumberReverser (int number) {
+   int rem, numTemp;
+   long long int rev = 0;
+   numTemp = number;
+   while (numTemp != 0) {
+      rem = numTemp % 10;
+      rev = rev * 10 + rem;
+      numTemp /= 10;
+   }
+   printf ("\nReversed: %lld", rev);
+   if (number >= 0 && number == rev)
+      return PALINDROME;
+   else
+      return NOTPALINDROME;
+}
+
+/// <summary>
+/// Function to run test cases
+/// </summary>
+void TestCases (int* numInputs, int numLength, char* strInputs[], int strLength, int* numExpected, int* strExpected) {
+   int* strOutputs = NULL, * numOutputs = NULL;
+   do {
+      strOutputs = (int*)malloc (strLength * sizeof (int));
+   } while (strOutputs == NULL);
+   for (int i = 0; i < strLength; i++) {
+      printf ("\nInput: %s", strInputs[i]);
+      strOutputs[i] = PalindromeChecker (strInputs[i]);
+      printf ("\nExpected: %d", strExpected[i]);
+      printf ("\nOutput: %d", strOutputs[i]);
+      if (strOutputs[i] == strExpected[i])
+         printf (ANSI_COLOR_GREEN"\nPassed\n"ANSI_COLOR_RESET);
+      else
+         printf (ANSI_COLOR_RED"\nFailed\n"ANSI_COLOR_RESET);
+   }
+   do {
+      numOutputs = (int*)malloc (numLength * sizeof (int));
+   } while (numOutputs == NULL);
+   for (int i = 0; i < numLength; i++) {
+      printf ("\nInput: %d", numInputs[i]);
+      numOutputs[i] = NumberReverser (numInputs[i]);
+      printf ("\nExpected: %d", numExpected[i]);
+      printf ("\nOutput: %d", numOutputs[i]);
+      if (numOutputs[i] == numExpected[i])
+         printf (ANSI_COLOR_GREEN"\nPassed\n"ANSI_COLOR_RESET);
+      else
+         printf (ANSI_COLOR_RED"\nFailed\n"ANSI_COLOR_RESET);
+   }
+}
+
+int main () {
+   char ch = '\n', choice[3], * result, term;
+   while (1) {
+      printf ("Menu");
+      printf ("\n~~~~");
+      printf ("\n1. Palindrome Checker");
+      printf ("\n2. Number Reverser");
+      printf ("\n3. Run Test Cases");
+      printf ("\n\nEnter your choice: ");
+      fgets (choice, 3, stdin);
+      result = strchr (choice, ch);
+      if (result == NULL) {
+         printf ("Invalid Choice!\n\n");
+         for (; ; ) {
+            term = fgetc (stdin);
+            if (term == '\n')
+               break;
+         }
+      } else if (choice[0] == '1') {
+         char phrase[MAX];
+         printf ("\nPalindrome Checker");
+         printf ("\n~~~~~~~~~~~~~~~~~~");
+         printf ("\nEnter a string: ");
+         fgets (phrase, MAX, stdin);
+         result = strchr (phrase, ch);
+         if (result == NULL) {
+            printf ("Input limit exceeded!\n");
+            for (; ; ) {
+               term = fgetc (stdin);
+               if (term == '\n')
+                  break;
+            }
+         } else {
+            if (PalindromeChecker (phrase) == PALINDROME)
+               printf ("Palindrome\n");
+            else if (PalindromeChecker (phrase) == INVALID)
+               printf ("Enter a valid string!\n");

```

Comment:
Here you are calling the function multiple times for the same input value. Instead, you could call it and store the output in a variable and use it to check the condition and print the result respectively.